### PR TITLE
School Classes

### PR DESCRIPTION
**Pull request by renercrows**
## Create class Person with the following:
- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if person is of age or if they have permission from parents.
## Create class Student with the following:
Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".
## Create class Teacher with the following:
- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.
- Each class should be saved in a separate file.

### CODE REVIEW

File: student_class.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require_relative 'person_class'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def intialize(classroom)
```
Comment (cynthiainga) [inline student_class.rb:6]: - [ ] Your initialize method does not call the super method with all the required methods, please kindly add the name, age, and parent_permission parameters.
eg:

```ruby
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher_class.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require_relative 'person_class'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization)
```
Comment (cynthiainga) [inline teacher_class.rb:6]: - [ ] Same remark about the parameters passed in initialize method, please kindly fix it as well 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @renercrows, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Good commit history ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student_class.rb (hunk 1 above)
Comment (renercrows) [reply student_class.rb]: I fixed the inittalizers.

File: teacher_class.rb (hunk 2 above)
Comment (renercrows) [reply teacher_class.rb]: I fixed this initializer too.

**Review by Reem-lab — APPROVED** [review]
# Hi @renercrows,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🔥🔥💯💯



## Highlights ✨✨
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the functionality✅
📌 Good commit messages ✅

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**